modules/auxiliary/scanner/http/nagios_xi_scanner.rb

```diff
@@ -0,0 +1,214 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HTTP::NagiosXi
+  include Msf::Auxiliary::Scanner
+  include Msf::Auxiliary::Report
+
+  def initialize
+    super(
+      'Name' => 'Nagios XI Scanner',
+      'Description' => %q{
+        The module detects the version of Nagios XI applications and
+        suggests matching exploit modules based on the version number.
+        Since Nagios XI applications only reveal the version to authenticated
+        users, valid credentials for a Nagios XI account are required.
+        Alternatively, it is possible to provide a specific Nagios XI version
+        number via the `VERSION` option. In that case, the module simply
+        suggests matching exploit modules and does not probe the target(s).
+
+      },
+      'Author' => [ 'Erik Wynter' ], # @wyntererik
+      'License' => MSF_LICENSE,
+      'References' =>
+        [
+          ['CVE', '2019-15949'],
+          ['CVE', '2020-5791'],
+          ['CVE', '2020-5792'],
+          ['CVE', '2020-35578']
+        ]
+    )
+    register_options [
+      OptString.new('VERSION', [false, 'Nagios XI version to check against existing exploit modules', nil])
+    ]
+  end
+
+  def finish_install
+    datastore['FINISH_INSTALL']
+  end
+
+  def username
+    datastore['USERNAME']
+  end
+
+  def password
+    datastore['PASSWORD']
+  end
+
+  def version
+    datastore['VERSION']
+  end
+
+  def parse_version(nagios_version)
+    # strip the xi- prefix used for Nagios XI versions in the official documentation
+    clean_version = nagios_version.downcase.delete_prefix('xi-')
+
+    # check for special case of 5r1.0, which is actually 5.1.0 but for some reason they still added the `r` used in legacy versions (this is the only version with this format)
+    if clean_version == '5r1.0'
+      clean_version = '5.1.0'
+    end
+
+    # Since we are dealing with a user provided version here, a strict regex check seems appropriate
+    # So far all modern Nagios XI versions start with 5. This regex can be updated if Nagios XI 6 ever comes out and modules for this new version are added
+    modern_check = clean_version.scan(/5\.\d{1,2}\.\d{1,2}/).flatten.first
+
+    unless modern_check.blank?
+      return ['modern', clean_version]
+    end
+
+    # Nagios XI legacy version numbers vary a lot, but they always start with a year followed by `r` and a single digit
+    legacy_check = clean_version.scan(/(20\d{2}r\d{1})/).flatten.first
+
+    unless legacy_check.blank?
+      return ['legacy', clean_version]
+    end
+
+    return 'unsupported'
+
+  end
+
+  def rce_check(version, real_target = nil)
```

Comment:
`real_target` is set to a boolean value in your code so its probably more appropriate to set it to `false` by default.

```suggestion
  def rce_check(version, real_target = false)
```

Reply:
actually had to change this to `rce_check(version, real_target: false)` otherwise Rubocop complained